### PR TITLE
Fix local fog volumes z axis calculation

### PR DESCRIPTION
**Pull request by RemyUnity**
---
### Purpose of this PR
The local volumetric fog was sampling 3D texture with an inverted Z axis. This PR fixes it.

Before: 
Before

After:
After

This change needed to compensate also the scrolling direction in Z.
For a scrolling speed of (0.5, 0.5, 0.5):
Before the PR:
af9538053f160788a72281f1f6854a54

After Z axis inversion:
4054395a48304c7c0a0947a151ec473c

After Z axis inversion and scroll direction correction:
e0e98c8fb881f06966437044fdf3e95e

---

### Testing status
Tested manually with [this project](https://drive.google.com/file/d/17Jkfhc4t1chWeX85HYvGdZvcBtNilFj4/view?usp=sharing), resulting to the screen captures above.

Images and test scene has been updated, but automation is still falling for reasons unrelated to my changes :/ : https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/hd%252Ffix%252Flocal-volumetric-fog-z-axis/.yamato%252Fall-hdrp.yml%2523Nightly_HDRP_trunk/9533137/job

Consoles PR: https://github.cds.internal.unity3d.com/unity/ScriptableRenderPipelinePrivate/pull/300

---
### Comments to reviewers
Pretty small change, but will alter the rendering of existing projects if they use the feature. Might be totally invisible in the case of noise 3D textures, but more problematic with specifically created textures.
Informations will be added in upgrade guides.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/LocalVolumetricFog.cs [hunk 1]
```diff
@@ -94,10 +94,8 @@ internal void Update(float time)
             //Update scrolling based on deltaTime
             if (volumeMask != null)
             {
-                textureOffset = (textureScrollingSpeed * time);
                 // Switch from right-handed to left-handed coordinate system.
-                textureOffset.x = -textureOffset.x;
-                textureOffset.y = -textureOffset.y;
+                textureOffset = -(textureScrollingSpeed * time);
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/LocalVolumetricFog.cs:98]: isn't this exactly the same code?

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/LocalVolumetricFog.cs:98]: ah right :)

**Review by sebastienlagarde — APPROVED** [review]

**Comment by kecho** [comment]
I still some dx12 errors, did you manage to figure out if they are related to the PR at all?

**Comment by sebastienlagarde** [comment]
crash isn't related to the PR (the DX12 one), DX12 was green on a previous run: https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/hd%252Ffix%252Flocal-volumetric-fog-z-axis/.yamato%252Fhdrp-win-dx12.yml%2523HDRP_Win_DX12_playmode_XR_mono_Linear_trunk/9525830/job . Merging

**Pull request merged**